src/validation.cpp

```diff
@@ -3396,22 +3396,21 @@ bool IsWitnessEnabled(const CBlockIndex* pindexPrev, const Consensus::Params& pa
 
 int GetWitnessCommitmentIndex(const CBlock& block)
 {
-    int commitpos = -1;
     if (!block.vtx.empty()) {
-        for (size_t o = 0; o < block.vtx[0]->vout.size(); o++) {
-            const CTxOut& vout = block.vtx[0]->vout[o];
+        for (size_t o = block.vtx[0]->vout.size(); o > 0; --o) {
+            const CTxOut& vout = block.vtx[0]->vout[o - 1];
             if (vout.scriptPubKey.size() >= MINIMUM_WITNESS_COMMITMENT &&
                 vout.scriptPubKey[0] == OP_RETURN &&
                 vout.scriptPubKey[1] == 0x24 &&
                 vout.scriptPubKey[2] == 0xaa &&
                 vout.scriptPubKey[3] == 0x21 &&
                 vout.scriptPubKey[4] == 0xa9 &&
                 vout.scriptPubKey[5] == 0xed) {
-                commitpos = o;
+                return o;
```

Comment:
`return o-1;` ?

Reply:
Yup yup, forgot to fix this. Initially was bad because `o` is signed. Will fix.